code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm

```diff
@@ -135,7 +136,7 @@
 		set_datum(FALSE)
 		hive.xenos_by_upgrade[upgrade] += src
 		update_wounds()
-		update_icon()
+		update_icon_state()
```

Comment:
```suggestion
		update_icon()
```
You're only supposed to change it in the proc that changes the icon states, not where you're calling it. Update_icon is meant to handle calling update_icon_state for you.